tests/eval/test_2fa_duo.py

```diff
@@ -0,0 +1,408 @@
+        fresh_browser.send_inputs(netid3, password)
+        fresh_browser.click(Locators.submit_button)
+        fresh_browser.wait_for_tag('p', 'Use your 2FA device.')
+        fresh_browser.find_element_by_name('rememberme').click()
+        fresh_browser.switch_to.frame(fresh_browser.find_element_by_id('duo_iframe'))
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        passcode = secrets.test_accounts.duo_code
+        fresh_browser.execute_script("document.getElementsByClassName('passcode-input')[0].value=arguments[0]", passcode)
+        fresh_browser.snap()
+        fresh_browser.execute_script("document.getElementById('passcode').click()")
+        fresh_browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
+        # go to an idp site to retrieve the rememberme cookie
+        fresh_browser.get('https://idp-eval.u.washington.edu/idp')
+        fresh_browser.wait_for_tag('h1', 'not found')
+
+        # look for these cookies:
+        # shib_idp_session, shib_idp_session_ss, and uw-rememberme-sptest03
+        cookie_list = fresh_browser.get_cookies()
+        wanted_cookie_names = ['shib_idp_session', 'shib_idp_session_ss', 'uw-rememberme-sptest03']
+
+        found_cookies = {}
+        for cookie in cookie_list:
+            if cookie['name'] in wanted_cookie_names:
+                found_cookies[cookie['name']] = 1
+
+        assert(len(found_cookies) == 3)
+
+        # b Sign out of the IdP to clear the 2FA session
+        fresh_browser.get('https://diafine6.sandbox.iam.s.uw.edu/Shibboleth.sso/Logout?return=https://idp-eval.u'
+                          '.washington.edu/idp/profile/Logout')
+        # After signing out of the IdP, the rememberme cookie should remain in the browser,
+        # but shib_idp_session* cookies should be gone.
+        fresh_browser.wait_for_tag('span', 'Your UW NetID sign-in session has ended.')
+
+        sign_out_cookie_list = fresh_browser.get_cookies()
+        found_rememberme = False
+        for cookie in sign_out_cookie_list:
+            if cookie['name'] == 'uw-rememberme-sptest03':
+                found_rememberme = True
+            else:
+                assert cookie['name'] not in wanted_cookie_names
+        assert found_rememberme is True
+
+        # c Navigate to the Eval "Token authn" link on (https://diafine12.sandbox.iam.s.uw.edu/shibevalmfa)
+        sp = ServiceProviderInstance.diafine12
+        with utils.using_test_sp(sp):
+            fresh_browser.get(f'{sp_url(sp)}/shibevalmfa')
+            password = secrets.test_accounts.password
+            fresh_browser.send_inputs(netid3, password)
+            fresh_browser.click(Locators.submit_button)
+            fresh_browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
+        fresh_browser.close()
+
+
+def test_forget_me_admin():
+    """
+    2FA-10 Forget me (admin) per Michael, this test is not yet ready to be automated.
+    """
```

Comment:
Please remove this. 

I think this either should be noted on the matrix that's being used as a source, or simply open a jira for this particular test for the future.